apps/vs-code-designer/src/app/commands/workflows/unitTest/saveBlankUnitTest.ts

```diff
@@ -2,29 +2,491 @@
+  unitTestFolderPath?: string;
+} {
+  const testsDirectoryUri = getTestsDirectory(projectPath);
+  const testsDirectory = testsDirectoryUri.fsPath;
+  // This logic for deriving logicAppName matches what generateBlankCodefulUnitTest currently uses
+  const logicAppName = path.basename(path.dirname(path.join(projectPath, workflowName)));
+  const logicAppFolderPath = path.join(testsDirectory, logicAppName);
+  const workflowFolderPath = path.join(logicAppFolderPath, workflowName);
+  const paths = {
+    testsDirectory,
+    logicAppName,
+    logicAppFolderPath,
+    workflowFolderPath,
+  };
+  if (unitTestName) {
+    paths['unitTestFolderPath'] = path.join(workflowFolderPath, unitTestName);
+  }
+  return paths;
+}
+
+/**
+ * Generates a C# class definition as a string.
+ * @param {string} logicAppName - The name of the Logic App, used as the namespace.
+ * @param {string} className - The name of the class to generate.
+ * @param {any} outputs - The outputs object containing properties to include in the class.
+ * @returns {string} - The generated C# class definition.
+ */
+function generateCSharpClasses(logicAppName: string, className: string, outputs: any): string {
+  // Start building the class definition
+  let classDefinition = 'using System;\nusing System.Collections.Generic;\nusing Newtonsoft.Json.Linq;\n\n';
+  classDefinition += `namespace ${logicAppName} {\n`;
+  classDefinition += `    public class ${className} {\n`;
+
+  // Generate properties for the class based on the outputs
+  for (const key in outputs) {
+    if (Object.prototype.hasOwnProperty.call(outputs, key)) {
+      const jsonType = outputs[key]?.type || 'object'; // Default to 'object' if type is not defined
+      const propertyType = mapJsonTypeToCSharp(jsonType);
+      const propertyName = toPascalCase(key);
+
+      // Add the property to the class definition
+      classDefinition += `        public ${propertyType} ${propertyName} { get; set; }\n`;
+    }
   }
+
+  // Close the class and namespace
+  classDefinition += '    }\n';
+  classDefinition += '}\n';
+
+  return classDefinition;
+}
+
+/**
+ * Maps JSON types to corresponding C# types.
+ * @param {string} jsonType - The JSON type (e.g., "string", "object", "array").
+ * @returns {string} - The corresponding C# type.
+ */
+function mapJsonTypeToCSharp(jsonType: string): string {
+  switch (jsonType) {
+    case 'string':
+      return 'string';
+    case 'integer':
+      return 'int';
+    case 'number':
+      return 'double';
+    case 'boolean':
+      return 'bool';
+    case 'array':
+      return 'List<object>';
+    case 'object':
+      return 'JObject';
+    case 'any':
+      return 'object';
+    case 'date-time':
+      return 'DateTime';
+    default:
+      return 'JObject';
+  }
+}
+
+/**
+ * Converts a string to PascalCase.
+ * @param {string} str - The input string.
+ * @returns {string} - The PascalCase version of the string.
+ */
+function toPascalCase(str: string): string {
```

Comment:
NIT: can you please move this function to this file? `libs/logic-apps-shared/src/utils/src/lib/helpers/stringFunctions.ts` this file mostly contains functions for strings. It would be great if you can add unit tests for it as well